File: myLeetCode/Dynamic_Programming/Basic/70. Climbing Stairs.cpp
// 70. Climbing Stairs
// https://leetcode.com/problems/climbing-stairs/

#include <iostream>
#include <vector>

int climbStairs(int n) {
    if (n < 0) return -1;
    std::vector<int> dp(n + 1, 0);
    dp[0] = 1;
    dp[1] = 1;

    for (int i = 2; i <= n; i++) {
        dp[i] = dp[i - 1] + dp[i - 2];
    }

    return dp[n];
}

int main(){
    std::cout << climbStairs(2) << std::endl; // print the output: 2

    std::cout << climbStairs(3) << std::endl; // print the output: 3

    std::cout << climbStairs(4) << std::endl; // print the output: 5

    return 0;
}

File: C++/代码随想录/Dynamic_Programming/Basic/70. Climbing Stairs.py
def climbStairs(n):
    if n <= 2:
        return n

    dp = [0] * (n + 1)
    dp[1] = 1
    dp[2] = 2

    for i in range(3, n + 1):
        dp[i] = dp[i - 1] + dp[i - 2]

    return dp[n]

# 测试用例
print(climbStairs(2))  # 应该输出 2
print(climbStairs(3))  # 应该输出 3
print(climbStairs(4))  # 应该输出 5
